### PR TITLE
MC-29345: ENGCOM: Update documentation (devdocs) according to 2.4

### PR DESCRIPTION
**Pull request by lenaorobei**
## Purpose of this pull request

Internal ticket MC-29345
- updated devdocs to reflect the current contribution process
- created symlink from 2.2. file to 2.3, because they contain the same info
- removed section about porting, since PRs will be accepted to 2.4-develop branch only

## Affected DevDocs pages
https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html
https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing.html

whatsnew
Updated the [Contributor Guide](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing.html) with guidance about accepting contributions to the 2.4 release line only.

### CODE REVIEW

**Comment by devops-devdocs** [comment]
An admin must run tests on this PR before it can be merged.

**Comment by dobooth** [comment]
Hi @lenaorobei. Thanks for the update. Please update the symlink for v2.3. It points to your local machine.

File: src/_includes/contributor/2-4-submission.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{:.bs-callout .bs-callout-info}
+**v2.4 Contribution Notice:**
+We no longer accept pull requests to the v 2.3, 2.2, v2.1 or v2.0 release lines to focus all development efforts on v2.4.
```
Comment (jeff-matthews) [inline src/_includes/contributor/2-4-submission.md]:
```suggestion
Magento is focusing development efforts on v2.4, so we are no longer accepting pull requests to the v2.3, v2.2, v2.1, and v2.0 release lines.
```

**Comment by lenaorobei** [comment]
@jeff-matthews, oh, my bad. I missed this.

**Comment by lenaorobei** [comment]
Please go ahead and merge! Thank you!

**Comment by jeff-matthews** [comment]
running tests

**Comment by jeff-matthews** [comment]
Tests failed. It looks like it has something to do with the symlink. I'm researching it.

```
Generating... 
/usr/local/bundle/gems/jekyll-4.0.0/lib/jekyll/utils.rb:141:in `initialize': No such file or directory @ rb_sysopen - /apps/slave/workspace/devdocs-testing/src/guides/v2.3/contributor-guide/contributing.md (Errno::ENOENT)
	from /usr/local/bundle/gems/jekyll-4.0.0/lib/jekyll/utils.rb:141:in `open'
	from /usr/local/bundle/gems/jekyll-4.0.0/lib/jekyll/utils.rb:141:in `has_yaml_header?'
	from /usr/local/bundle/gems/jekyll-4.0.0/lib/jekyll/reader.rb:52:in `block in read_directories'
	from /usr/local/bundle/gems/jekyll-4.0.0/lib/jekyll/reader.rb:48:in `each'
	from /usr/local/bundle/gems/jekyll-4.0.0/lib/jekyll/reader.rb:48:in `read_directories'
	from /usr/local/bundle/gems/jekyll-4.0.0/lib/jekyll/reader.rb:89:in `block in retrieve_dirs'
	from /usr/local/bundle/gems/jekyll-4.0.0/lib/jekyll/reader.rb:86:in `each'
	from /usr/local/bundle/gems/jekyll-4.0.0/lib/jekyll/reader.rb:86:in `retrieve_dirs'
	from /usr/local/bundle/gems/jekyll-4.0.0/lib/jekyll/reader.rb:60:in `read_directories'
	from /usr/local/bundle/gems/jekyll-4.0.0/lib/jekyll/reader.rb:89:in `block in retrieve_dirs'
	from /usr/local/bundle/gems/jekyll-4.0.0/lib/jekyll/reader.rb:86:in `each'
	from /usr/local/bundle/gems/jekyll-4.0.0/lib/jekyll/reader.rb:86:in `retrieve_dirs'
	from /usr/local/bundle/gems/jekyll-4.0.0/lib/jekyll/reader.rb:60:in `read_directories'
	from /usr/local/bundle/gems/jekyll-4.0.0/lib/jekyll/reader.rb:89:in `block in retrieve_dirs'
	from /usr/local/bundle/gems/jekyll-4.0.0/lib/jekyll/reader.rb:86:in `each'
	from /usr/local/bundle/gems/jekyll-4.0.0/lib/jekyll/reader.rb:86:in `retrieve_dirs'
	from /usr/local/bundle/gems/jekyll-4.0.0/lib/jekyll/reader.rb:60:in `read_directories'
	from /usr/local/bundle/gems/jekyll-4.0.0/lib/jekyll/reader.rb:16:in `read'
	from /usr/local/bundle/gems/jekyll-4.0.0/lib/jekyll/site.rb:174:in `read'
	from /usr/local/bundle/gems/jekyll-4.0.0/lib/jekyll/site.rb:74:in `process'
	from /usr/local/bundle/gems/jekyll-4.0.0/lib/jekyll/command.rb:28:in `process_site'
	from /usr/local/bundle/gems/jekyll-4.0.0/lib/jekyll/commands/build.rb:65:in `build'
	from /usr/local/bundle/gems/jekyll-4.0.0/lib/jekyll/commands/build.rb:36:in `process'
	from /usr/local/bundle/gems/jekyll-4.0.0/lib/jekyll/command.rb:89:in `block in process_with_graceful_fail'
	from /usr/local/bundle/gems/jekyll-4.0.0/lib/jekyll/command.rb:89:in `each'
	from /usr/local/bundle/gems/jekyll-4.0.0/lib/jekyll/command.rb:89:in `process_with_graceful_fail'
	from /usr/local/bundle/gems/jekyll-4.0.0/lib/jekyll/commands/build.rb:18:in `block (2 levels) in init_with_program'
	from /usr/local/bundle/gems/mercenary-0.3.6/lib/mercenary/command.rb:220:in `block in execute'
	from /usr/local/bundle/gems/mercenary-0.3.6/lib/mercenary/command.rb:220:in `each'
	from /usr/local/bundle/gems/mercenary-0.3.6/lib/mercenary/command.rb:220:in `execute'
	from /usr/local/bundle/gems/mercenary-0.3.6/lib/mercenary/program.rb:42:in `go'
	from /usr/local/bundle/gems/mercenary-0.3.6/lib/mercenary.rb:19:in `program'
	from /usr/local/bundle/gems/jekyll-4.0.0/exe/jekyll:15:in `<top (required)>'
	from bin/jekyll:16:in `load'
	from bin/jekyll:16:in `<main>'
rake aborted!
Command failed with status (1): [bin/jekyll build --config=_config.yml,_con...]
/apps/slave/workspace/devdocs-testing/rakelib/build.rake:6:in `block (2 levels) in <top (required)>'
Tasks: TOP => build:prod
```

**Comment by jeff-matthews** [comment]
@lenaorobei, will you please remove the symlink and relink it? I did this locally and it resolved the error.

```
git rm src/guides/v2.3/contributor-guide/contributing.md
cd src/guides/v2.3/contributor-guide
ln -s ../../../guides/v2.2/contributor-guide/contributing.md
```

**Comment by lenaorobei** [comment]
@jeff-matthews resolved conflicts and pushed

**Comment by jeff-matthews** [comment]
running tests

**Comment by jeff-matthews** [comment]
running tests

**Comment by jeff-matthews** [comment]
Test failed on broken internal links. Here's an excerpt from the logs:

```
+ rake test:html
[35mChecking links with html-proofer...[0m
Running ["HtmlCheck", "ImageCheck", "ScriptCheck", "LinkCheck", "LinkChecker::DoubleSlashCheck"] on ["_site"] on *.html... 

Ran on 2873 files!

- internally linking to /guides/v2.3/contributor-guide/contributing.html, which does not exist
  *  _site/404.html (line 134)
     <a href="/guides/v2.3/contributor-guide/contributing.html">Contributor Guide</a>
  *  _site/codelinks/attributes.html (line 134)
     <a href="/guides/v2.3/contributor-guide/contributing.html">Contributor Guide</a>
  *  _site/community/resources/best-practices.html (line 134)
     <a href="/guides/v2.3/contributor-guide/contributing.html">Contributor Guide</a>
...
rake aborted!
HTML-Proofer found 1317 failures!
```

The large number of errors for a single file indicates that it's an error in the table of contents for the same file. I'm not sure why it can't find the `/guides/v2.3/contributor-guide/contributing.html` file, but I'm looking into it.

**Comment by lenaorobei** [comment]
Maybe it is related to resolving conflicts and symlink became broken? Let me try to create it one more time.

**Comment by jeff-matthews** [comment]
Oh maybe. I'll pull down your branch locally and run tests to verify.

**Comment by jeff-matthews** [comment]
That's much better! But there are still a couple of errors:

```
- linking to internal hash #porting that does not exist
  *  _site/guides/v2.2/contributor-guide/contributing.html (line 270)
     <a href="#porting">Porting code contributions across Magento versions</a>
  *  _site/guides/v2.3/contributor-guide/contributing.html (line 265)
     <a href="#porting">Porting code contributions across Magento versions</a>
```

**Comment by lenaorobei** [comment]
@jeff-matthews fixed

File: src/_includes/contributor/labels.md [hunk 2]
```diff
@@ -1,10 +1,10 @@
 ### Release Lines
 {:.no_toc}
 
-Release line labels indicate the specific Magento release lines affected by the issue or PR. For example, if working on a fix for 2.2.6, you would apply the Release Line: 2.2. This effectively includes all releases in this line.
+Release line labels indicate the specific Magento release lines affected by the issue or PR. For example, if working on a fix for 2.4.1 you would apply the Release Line: 2.4. This effectively includes all releases in this line.
```
Comment (keharper) [inline src/_includes/contributor/labels.md]: 2.4.1 doesn't currently make sense. Change to 2.4.0?

**Comment by jeff-matthews** [comment]
running tests

**Pull request merged**